quickwit-indexing/src/source/kafka_source.rs

```diff
@@ -172,65 +172,76 @@ impl Source for KafkaSource {
         batch_sink: &Mailbox<Indexer>,
         ctx: &SourceContext,
     ) -> Result<(), ActorExitStatus> {
+        let mut batch_num_bytes = 0;
         let mut docs = Vec::new();
         let mut checkpoint_delta = CheckpointDelta::default();
 
-        let deadline = tokio::time::sleep(quickwit_actors::HEARTBEAT / 2);
-        let mut message_stream = Box::pin(self.consumer.stream().take_until(deadline));
-
-        let mut batch_num_bytes = 0;
-
-        while let Some(message_res) = message_stream.next().await {
-            let message = match message_res {
-                Ok(message) => message,
-                Err(KafkaError::PartitionEOF(partition_id)) => {
-                    self.state.num_active_partitions -= 1;
-                    info!(
-                        topic = %self.topic,
-                        partition_id = ?partition_id,
-                        num_active_partitions = ?self.state.num_active_partitions,
-                        "Reached end of partition."
-                    );
-                    continue;
+        let deadline = tokio::time::sleep(quickwit_actors::HEARTBEAT);
+        tokio::pin!(deadline);
+
+        let mut stream = self.consumer.stream();
+
+        loop {
+            tokio::select! {
+                message_opt = stream.next() => {
+                    if let Some(message_res) = message_opt {
+                        let message = match message_res {
```

Comment:
I think the indentation reaches a level here where is hurts readability

Reply:
This is relevant now since we're not going to move forward with this PR as is, but I agree with you :)